File: frontend/famcare_app/lib/models/slot.dart
import 'caregiver.dart';

class AvailableSlot {
  final String startTime;
  final String endTime;
  final List<Caregiver> availableCaregivers;

  const AvailableSlot({
    required this.startTime,
    required this.endTime,
    required this.availableCaregivers,
  });

  factory AvailableSlot.fromJson(Map<String, dynamic> json) {
    return AvailableSlot(
      startTime: json['start_time'] as String,
      endTime: json['end_time'] as String,
      availableCaregivers: (json['available_caregivers'] as List)
          .map((c) => Caregiver.fromJson(c as Map<String, dynamic>))
          .toList(),
    );
  }
}

class SlotAvailability {
  final String serviceId;
  final String serviceName;
  final int durationMinutes;
  final String date;
  final List<AvailableSlot> slots;

  const SlotAvailability({
    required this.serviceId,
    required this.serviceName,
    required this.durationMinutes,
    required this.date,
    required this.slots,
  });

  factory SlotAvailability.fromJson(Map<String, dynamic> json) {
    return SlotAvailability(
      serviceId: json['service_id'] as String,
      serviceName: json['service_name'] as String,
      durationMinutes: json['duration_minutes'] as int,
      date: json['date'] as String,
      slots: (json['slots'] as List)
          .map((s) => AvailableSlot.fromJson(s as Map<String, dynamic>))
          .toList(),
    );
  }
}

File: backend/app/schemas/slot.py
"""Slot availability schemas."""

from typing import List
from pydantic import BaseModel


class SlotAvailabilityRequest(BaseModel):
    """Query parameters for slot availability."""
    service_id: str
    date: str  # "YYYY-MM-DD"


class AvailableCaregiver(BaseModel):
    """A caregiver available for a specific slot."""
    id: str
    name: str


class SlotResponse(BaseModel):
    """A single available time slot with its available caregivers."""
    start_time: str       # "HH:MM"
    end_time: str         # "HH:MM"
    available_caregivers: List[AvailableCaregiver]


class SlotAvailabilityResponse(BaseModel):
    """Full availability response for a service on a date."""
    service_id: str
    service_name: str
    duration_minutes: int
    date: str
    slots: List[SlotResponse]
